api/client/client.go

```diff
@@ -111,19 +108,18 @@ type Config struct {
 	// KeepAliveCount specifies the amount of missed keep alives
 	// to wait for before declaring the connection as broken
 	KeepAliveCount int
-	// TLS is the client's TLS config
-	TLS *tls.Config
+	// Credentials are used to authenticate the client's connection to the server
+	Credentials
```

Comment:
@Joerger I think the problem with separating the credentials from the rest of the client configuration is that some functions, like `LoadProfile` will give you both `*tls.Config` and `Addrs` and separating credentials from configuration makes that a bit cumbersome to use.

Below is a start at this, maybe you have some better ideas as well?

```go
// FromProfile is a convenience function that configures a client from the
// current profile. It is functionally equivalent to the following:
//      
// cfg, err := LoadProfile()     
// if err != nil {
//   return err
// }    
// clt, err := client.New(cfg)
// if err != nil {
//   return err
// }
func FromProfile() (*Client, error)
    
// LoadProfile creates client configuration from the current profile. Configures
// the x509 credentials and dialer. 
func LoadProfile() (*client.Config, error)

// LoadIdentityFile loads client configuration from an identity file. This
// function only loads the x509 credentials, the caller is still required to
// configure the addresses (or a dialer).
//
// cfg, err := client.LoadIdentityFile("/path/to/identity.pem")
// if err != nil {
//    return err
// }
// cfg.Addrs = []string{"auth.example.com:3025"}
//
// clt, err := client.New(cfg)
// if err != nil {
//    return err
// }
func LoadIdentityFile(identityFile string) (*client.Config, error)

// LoadKeyPair loads client configuration from a key pair on disk. This
// function only loads the x509 credentials, the caller is still required to
// configure the addresses (or a dialer).
func LoadKeyPair(certFile string, keyFile string) (*client.Config, error)

// LoadConfig loads client configuration from a *tls.Config. This
// function only loads the x509 credentials, the caller is still required to
// configure the addresses (or a dialer).
func LoadConfig(*tls.Config) (*client.Config, error)
```